### PR TITLE
Style swap step in aggregator

### PR DESCRIPTION
**Pull request by th0rOdinson**

### CODE REVIEW

File: apps/root/src/common/components/transaction-steps/index.tsx [hunk 1]
```diff
@@ -170,6 +171,23 @@ export type TransactionAction =
...
+type TransactionActionProps =
+  | TransactionActionApproveTokenProps
+  | TransactionActionApproveTokenSignProps
+  | TransactionActionWaitForApprovalProps
+  | TransactionActionWaitForSimulationProps
+  | TransactionActionWaitForQuotesSimulationProps
+  | TransactionActionSwapProps
+  | TransactionActionCreatePositionProps;
+
+type CommonTransactionActionProps = Pick<
```
Comment (FiboApe) [inline apps/root/src/common/components/transaction-steps/index.tsx:183]: wouldn't this just be ItemProps?

File: apps/root/src/common/components/transaction-steps/index.tsx [hunk 2]
```diff
@@ -229,6 +250,63 @@ const StyledTransactionStepButtonContainer = styled.div`
...
+  ${({ theme: { palette } }) => `
+  color: ${colors[palette.mode].typography.typo3};
+  `}
+`;
+
+const StyledTickCircleIcon = styled(TickCircleIcon)`
+  color: ${({ theme: { palette } }) => colors[palette.mode].semantic.success.darker};
+`;
+
+const CommonTransactionStepItem = ({
```
Comment (FiboApe) [inline apps/root/src/common/components/transaction-steps/index.tsx:270]: Oh this is cool

File: apps/root/src/common/components/transaction-steps/index.tsx [hunk 3]
```diff
@@ -333,7 +407,7 @@ const buildApproveTokenItem = ({
               >
                 {hasPendingApproval ? waitingForAppvText : isPermit2Enabled ? infiniteBtnText : specificBtnText}
               </Button>
-              {!isPermit2Enabled && (
+              {isPermit2Enabled && (
```
Comment (FiboApe) [inline apps/root/src/common/components/transaction-steps/index.tsx:410]: was this a bug that we had?

File: apps/root/src/common/components/transaction-steps/index.tsx [hunk 4]
```diff
@@ -346,32 +420,41 @@ const buildApproveTokenItem = ({
...
+                  <FormattedMessage description="viewExplorer" defaultMessage="View in explorer" />
+                )}
+              </Button>
+              {!isPendingTransaction && receipt && (
+                <ContainerBox gap={2} alignItems="center">
+                  <StyledTickCircleIcon />
+                  <Typography variant="body" fontWeight="600">
+                    <FormattedMessage
+                      description="approveSellAmount"
+                      defaultMessage="Sell {amount} {symbol}"
```
Comment (FiboApe) [inline apps/root/src/common/components/transaction-steps/index.tsx:446]: Is this correct? Isn't this the approval step?

File: apps/root/src/common/components/transaction-steps/index.tsx (hunk 3 above)
Comment (th0rOdinson) [reply apps/root/src/common/components/transaction-steps/index.tsx:410]: Indeed, after testing this once again, I found out that this button wasn't rendering in the right conditions.
Now, this secondary button (exact amount approval) is only rendered when permit2 is enabled

File: apps/root/src/common/components/transaction-steps/index.tsx (hunk 1 above)
Comment (th0rOdinson) [reply apps/root/src/common/components/transaction-steps/index.tsx:183]: Properties such as `explanation` and `done` props are related to TransactionActionProps and not present in ItemProps. By using TransactionActionProps (extended from ItemProps) I can pick those two and also  `isLast` and `isCurrentStep` from ItemProps.
We could redefine the scope of both interfaces but I found this to be the easiest way to proceed

File: apps/root/src/common/components/transaction-steps/index.tsx (hunk 4 above)
Comment (th0rOdinson) [reply apps/root/src/common/components/transaction-steps/index.tsx:446]: I found this label in the design, and it seemed prudent to me to add it, even though it was on another step. I should have informed you beforehand 😅 
Screenshot 2024-02-08 at 11 20 11

PS: After reviewing it, I think it should be "Approve 20 NEXT"

File: apps/root/src/common/components/transaction-steps/index.tsx (hunk 1 above)
Comment (FiboApe) [reply apps/root/src/common/components/transaction-steps/index.tsx:183]: But ItemProps is only used to extend each actionProp so shouldn't done and explanation be there?

File: apps/root/src/common/components/transaction-steps/index.tsx (hunk 3 above)
Comment (FiboApe) [reply apps/root/src/common/components/transaction-steps/index.tsx:410]: Exact approval should actually appear when permit2 is disabled, now that I'm thinking about it we should always show exact approvals

File: apps/root/src/common/components/transaction-steps/index.tsx (hunk 4 above)
Comment (FiboApe) [reply apps/root/src/common/components/transaction-steps/index.tsx:446]: Yup it should be approve

File: apps/root/src/common/components/transaction-steps/index.tsx (hunk 1 above)
Comment (th0rOdinson) [reply apps/root/src/common/components/transaction-steps/index.tsx:183]: Oh I think I get it now!

File: apps/root/src/common/components/transaction-steps/index.tsx (hunk 3 above)
Comment (th0rOdinson) [reply apps/root/src/common/components/transaction-steps/index.tsx:410]: Agree, maybe I didn't explain myself clearly, I believe this is implemented as expected:

### Permit2 Enabled
Screenshot 2024-02-08 at 16 52 10

### Permit2 Disabled
Screenshot 2024-02-08 at 16 55 18

Comment (FiboApe) [reply apps/root/src/common/components/transaction-steps/index.tsx:409]: Gotcha

**Pull request merged**